### PR TITLE
Fixes Typecheck bug in preactjs/preact#3934 

### PR DESCRIPTION
**Pull request by paulreece**
Added SignalLike as an option for default that fixes [/preact/issues/3934](https://github.com/preactjs/preact/issues/3934) while preserving current functionality.  This is a simple fix but it seems like it would do what's needed.

### CODE REVIEW

**Comment by paulreece** [comment]
@marvinhagemeister @rschristian Let me know what you think and if any changes are required.

File: index.d.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface CustomHistory {
 
 export interface RoutableProps {
 	path?: string;
-	default?: boolean;
+	default?:  boolean | preact.SignalLike<boolean | undefined>;
```
Comment (rschristian) [inline index.d.ts:24]:
```suggestion
	default?: boolean | preact.JSX.SignalLike<boolean | undefined>;
```

**Review by rschristian — APPROVED** [review]
CI failure is due to us having an older version of Preact (without `SignalLike`) as a devDep, nothing you necessarily need to worry about though I don't love the idea of increasing the minimum supported Preact version (if the user wants TypeScript support, anyways).

**Comment by paulreece** [comment]
Thanks for the quick response!  Let me know if I can help more with this or anything else.

File: package.json [hunk 2]
```diff
@@ -84,25 +84,21 @@
 	"devDependencies": {
 		"@babel/plugin-transform-react-jsx": "^7.9.1",
 		"@types/jasmine": "^3.10.3",
-		"chai": "^3.5.0",
-		"copyfiles": "^1.0.0",
+		"chai": "^4.3.7",
```
Comment (rschristian) [inline package.json:87]: See #453 for justification, but need to update some of these here w/ the type fix to satisfy CI.

**Comment by rschristian** [comment]
We really appreciate the fix! ❤️

I'll try to cut a release in a bit, after my dinner.

**Pull request merged**